src/main/kotlin/net/leanix/vsm/sbomBooster/service/BitBucketApiService.kt

```diff
@@ -0,0 +1,106 @@
+package net.leanix.vsm.sbomBooster.service
+
+import net.leanix.vsm.sbomBooster.configuration.PropertiesConfiguration
+import net.leanix.vsm.sbomBooster.domain.BitBucketAuthResponse
+import net.leanix.vsm.sbomBooster.domain.BitBucketRepositoriesResponse
+import net.leanix.vsm.sbomBooster.domain.GitProviderApiService
+import net.leanix.vsm.sbomBooster.domain.Repository
+import org.slf4j.LoggerFactory
+import org.springframework.http.HttpEntity
+import org.springframework.http.HttpHeaders
+import org.springframework.http.HttpMethod
+import org.springframework.http.MediaType
+import org.springframework.stereotype.Service
+import org.springframework.util.LinkedMultiValueMap
+import org.springframework.util.MultiValueMap
+import org.springframework.web.client.RestTemplate
+import java.util.*
+
+@Service
+class BitBucketApiService(
+    private val propertiesConfiguration: PropertiesConfiguration
+) : GitProviderApiService {
+
+    companion object {
+        private val logger = LoggerFactory.getLogger(BitBucketApiService::class.java)
```

Comment:
This was picked up by our static analysis tool as a violation because the logger variable is unused. 
You can run this locally by executing the gradle detekt or check task.